ffi/src/lib.rs

```diff
@@ -146,6 +250,18 @@ pub extern "C" fn store_open(uri: *const c_char) -> *mut Store {
 
 // TODO: begin_transaction
 
+/// Performs a single transaction against the store.
+///
+/// # Safety
+///
+/// Callers must ensure that the pointer to the `Store` is not dangling and that
+/// the C string provided to `transaction` is valid.
+///
+/// Callers are responsible for managing the memory for the return value.
+/// A destructor `destroy` is provided for releasing the memory for this
+/// pointer type.
+///
+/// TODO: Document the errors that can result from transact
```

Comment:
One reason this is hard is that the transactor uses `error-chain` -- badly.  Most of the errors are string values, which of course isn't helpful.  Making this better requires work, which might be part of #586, or that I might do independently.